presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java

```diff
@@ -287,6 +287,12 @@ private void updatePeakMemoryReservations()
         peakTotalMemoryReservation.accumulateAndGet(totalMemory, Math::max);
     }
 
+    // listen to revocable memory allocations and call any listeners waiting on task memory allocation
+    private void updateTaskRevocableMemoryReservation()
+    {
+        driverContext.getPipelineContext().getTaskContext().getQueryContext().getMemoryPool().onTaskMemoryReserved(driverContext.getTaskId());
```

Comment:
Not too sure what we think of this way of alerting the TaskThresholdMemoryRevokingScheduler. Not ideal but this is the cleanest way I could think of - other alternatives are passing the listener around somehow instead of piggybacking on MemoryPool. The thing I dont like about this is that MemoryPool now has two sets of disparate listeners and technically this is a task/operator context allocation.